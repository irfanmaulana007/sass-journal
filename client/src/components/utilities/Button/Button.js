import React from 'react'
import PropTypes from "prop-types"

function handleVariant(variant) {
    let variantClassName
    switch(variant) {
    case 'primary': variantClassName = 'btn-primary'
        break;
    case 'success': variantClassName = 'btn-success'
        break;
    case 'danger': variantClassName = 'btn-danger'
        break;
    default: variantClassName = 'btn-theme'
    }

    return variantClassName
}

function handleSize(size) {
    let sizeClassName
    switch(size) {
    case 'wide': sizeClassName = 'btn-wide'
        break;
    case 'block': sizeClassName = 'btn-block'
        break;
    default: sizeClassName = ''
    }

    return sizeClassName
}

function Button ({ variant, size, label, handleClick }) {
    return (
        <button onClick={handleClick} className={`btn ${handleVariant(variant)} ${handleSize(size)} text-capitalize`}>{label}</button>
    )
}

Button.propTypes = {
    label: PropTypes.string.isRequired,
    variant: PropTypes.string,
    size: PropTypes.string,
    handleClick: PropTypes.func
}

export default Button