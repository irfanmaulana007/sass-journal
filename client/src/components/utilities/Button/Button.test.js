import React from 'react'
import { shallow } from 'enzyme'

import Button from './Button'

const props = {
    label: 'Submit'
}
const wrapped = shallow(<Button label={props.label} debug />)

describe('Button', () => {
    it('should render the Button Component correctly', () => {
        expect(wrapped).toMatchSnapshot();
    });
    it('renders the Button Label', () => { 
        expect(wrapped.text()).toEqual(props.label);
    });
});