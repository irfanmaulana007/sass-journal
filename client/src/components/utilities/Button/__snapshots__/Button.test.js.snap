// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the Button Component correctly 1`] = `ShallowWrapper {}`;
